src/Microsoft.Azure.WebJobs.Host/Bindings/QueueBindingProvider.cs

```diff
@@ -0,0 +1,262 @@
+            IConverterManager converterManager)
+        {
+            // IStorageQueueMessage is the core testing interface 
+            converterManager.AddConverter<byte[], IStorageQueueMessage, QueueAttribute>(ConvertByteArrayToCloudQueueMessage);
+            converterManager.AddConverter<IStorageQueueMessage, byte[]>(ConvertCloudQueueMessageToByteArray);
+
+            converterManager.AddConverter<string, IStorageQueueMessage, QueueAttribute>(ConvertStringToCloudQueueMessage);
+            converterManager.AddConverter<IStorageQueueMessage, string>(ConvertCloudQueueMessageToString);
+
+            converterManager.AddConverter<CloudQueueMessage, IStorageQueueMessage>(ConvertToStorageQueueMessage);
+
+            var bindingFactory = new BindingFactory(nameResolver, converterManager);
+
+            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<QueueAttribute, IStorageQueueMessage>(BuildFromQueueAttribute, ToWriteParameterDescriptorForCollector, CollectAttributeInfo);
+            var bindClient = bindingFactory.BindToExactAsyncType<QueueAttribute, IStorageQueue>(BuildClientFromQueueAttributeAsync, ToReadWriteParameterDescriptorForCollector, CollectAttributeInfo);
+            var bindSdkClient = bindingFactory.BindToExactAsyncType<QueueAttribute, CloudQueue>(BuildRealClientFromQueueAttributeAsync, ToReadWriteParameterDescriptorForCollector, CollectAttributeInfo);
+                        
+            var bindingProvider = new GenericCompositeBindingProvider<QueueAttribute>(
+                ValidateQueueAttribute, nameResolver, bindClient, bindSdkClient, bindAsyncCollector);
+
+            return bindingProvider;
+        }
+
+        // [Queue] has some legacy behavior where the storage account can be specified outside of the [Queue] attribute. 
+        // The storage account is pulled from the ParameterInfo (which could pull in a [Storage] attribute on the container class)
+        // Resolve everything back down to a single attribute so we can use the binding helpers. 
+        // This pattern should be rare since other extensions can just keep everything directly on the primary attribute. 
+        private async Task<QueueAttribute> CollectAttributeInfo(QueueAttribute attrResolved, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            // Look for [Storage] attribute and squirrel over 
+            IStorageAccount account = await _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver);
+            StorageClientFactoryContext clientFactoryContext = new StorageClientFactoryContext
+            {
+                Parameter = parameter
+            };
+            IStorageQueueClient client = account.CreateQueueClient(clientFactoryContext);
+                    
+            return new ResolvedQueueAttribute(attrResolved.QueueName, client);
+        }
+
+        // Cloudqueue version. Can technically read from it. Preserves compat with older SDK. 
```

Comment:
I'll reword.

---
In reply to: [64935628](https://github.com/Azure/azure-webjobs-sdk/pull/711#discussion_r64935628) [](ancestors = 64935628)